plugins/cilium-cni/cmd/endpoint.go

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/api/v1/models"
+	ipamOption "github.com/cilium/cilium/pkg/ipam/option"
+	"github.com/cilium/cilium/plugins/cilium-cni/types"
+)
+
+// EndpointConfigurator returns a list of endpoint configurations for a given
+// CNI ADD invocation. If the CNI ADD invocation should result in multiple endpoints
+// being created, it may return multiple endpoint configurations, one for each endpoint.
+type EndpointConfigurator interface {
+	GetConfiguration(p ConfigurationParams) ([]EndpointConfiguration, error)
+}
+
+// EndpointConfiguration determines the configuration of an endpoint to be
+// created duing a CNI ADD invocation.
+type EndpointConfiguration interface {
+	// IfName specifies the container interface name to be used for this endpoint
+	IfName() string
+	// IPAMPool specifies what IPAM pool the endpoint's IP should be allocated from
+	IPAMPool() string
+
+	// PrepareEndpoint returns the template for the endpoint creation request ep,
+	// as well as the interface configuration cmd of the container namespace.
```

Comment:
nit: mention the return values in the same order as in the method signature return values

```suggestion
	// PrepareEndpoint returns the interface configuration cmd of the container namespace as well as the template for the endpoint creation request ep.
```